src/errorpage.cc

```diff
@@ -160,6 +221,11 @@ errorInitialize(void)
     const char *text;
     error_page_count = ERR_MAX + ErrorDynamicPages.size();
     error_text = static_cast<char **>(xcalloc(error_page_count, sizeof(char *)));
+    // error pages validator to use as base
+    ErrTextValidator validator("ErrorPageFile");
+    validator.warn(DBG_CRITICAL);
+    if (!reconfiguring)
```

Comment:
According to the [table](https://github.com/measurement-factory/squid/pull/4#issuecomment-401416167), in case of errors Squid should be killed only during startup. Should we change the condition to `if (starting_up)` then?

Reply:
Yes but on reconfiguration you do not want to kill squid for template errors. The "if (!reconfiguring)" warn me about the problematic "reconfiguring" case.

Reply:
TLDR: Please keep the `!reconfiguring` condition.

----

FWIW, the table was written before I knew about the existence of `-k restart`. The table does not cover that case explicitly. Implicitly, the intent should be clear: Ideally,

* initial configuration with malformed files should result in Squid startup failure/death;
* any reconfiguration with malformed files should result in no changes to configuration; the fewer side effects such reconfiguration has, the better; process deaths are certainly undesirable.

Due to lots of old, poorly designed code, we cannot implement the above ideal in this project alone, but that ideal is what we should gravitate towards when making choices today.

> should Squid be killed on restarting (-k restart option)? If no, then "starting_up" is mandatory in this context.

When Squid is performing reconfiguration, the `reconfiguring` variable should be set to true. Whether "restarting" involves such a reconfiguration step should be irrelevant. IIRC, restarting code does not involve reconfiguration today, but if that changes, setting `reconfiguring` to true should be a part of those changes.

If is possible that we will need to split `reconfiguring` into two variables/states:

1. Replacing one configuration with another.
2. Reacting to a "-k reconfigure" or equivalent. Includes (1) as a sub-task.

Until that split happens, `reconfiguring` may be interpreted as "Replacing one configuration with another" by the configuration parsing code. That configuration code, in general, should not be bothered with the knowledge of various "-k" commands and their effects.

I doubt such split is necessary during this issue.

Reply:
So I am keeping the "!reconfiguring" and marking this as resolved